freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -704,8 +639,28 @@ public TopologySegment deleteTopologySegment(String topologySuffixCn, TopologySe
         return (TopologySegment) invoke("topologysegment_del", flags, params, TopologySegment.class).getResult();
     }
 
+    public void callBatch(BiConsumer<String, String> warnings, List<Object> operations,
+            Set<FreeIpaErrorCodes> acceptableErrorCodes) throws FreeIpaClientException {
+        callBatch(warnings, operations, DEFAULT_BATCH_CALL_PARTITION_SIZE, acceptableErrorCodes);
+    }
+
+    public void callBatch(BiConsumer<String, String> warnings, List<Object> operations, Integer partitionSize,
+            Set<FreeIpaErrorCodes> acceptableErrorCodes) throws FreeIpaClientException {
+        List<List<Object>> partitions = Lists.partition(operations, partitionSize);
+        for (List<Object> operationsPartition : partitions) {
+            BatchOperation.get(operationsPartition, warnings, acceptableErrorCodes).invoke(this);
+        }
+    }
+
+    public static void checkIfClientStillUsable(FreeIpaClientException e) throws FreeIpaClientException {
+        if (e.isClientUnusable()) {
+            LOGGER.warn("Client is not usable for further usage");
```

Comment:
nit: sounds a bit strange reading this. I think sth like `Client is not usable anymore` should be enough, or similar

Reply:
makes sense, will change it